General/Pragmas.md

```diff
@@ -0,0 +1,182 @@
+- Select a symbol which will be the name of the pragma and right click on it then select `Browse senders` in the search menu.
+- Type the name of the pragma in the Spotter (`SHIFT + ENTER`) followed by `#p`.
+- Use the `Finder` (`Menubar -> Tools -> Finder`) and select the `Pragmas` mode.
+- Send the message `senders` to a symbol which is the name of the pragma.
+
+### Get the pragmas of a CompiledMethod
+
+If you have a compile method you can directly ask it his pragmas via the `#pragmas` method.
+
+For example, you can get all methods with a pragma like this:
+
+```Smalltalk
+SystemNavigation new browseAllSelect: [ :method | method pragmas isNotEmpty]
+```
+
+### Collect pragmas in a hierarchy
+
+When you want to use the pragmas in your application you have two ways to collect them. If you want to find the pragmas in one class or one hierarchy of class, you can use the `Pragma` class to collect them.
+
+The Pragma class implements different ways to get to pragmas depending on what you want.
+
+You can use:
+- `allNamed: aSymbol from: aSubClass to: aSuperClass`
+- `allNamed: aSymbol from: aSubClass to: aSuperClass sortedByArgument: anInteger`
+- `allNamed: aSymbol in: aClass`
+- `allNamed: aSymbol in: aClass sortedByArgument: anInteger`
+...
+
+In our example, we might want to use this way of accessing our pragmas if all parameters are defined in a configuration object.
+
+In that case, we can access them this way:
+
+```Smalltalk
+Pragma allNamed: #applicationParameterOptional: in: MyConfiguration
+```
+
+Or if we have a hierarchy of configurations:
+
+```Smalltalk
+Pragma allNamed: #applicationParameterOptional: from: MyConfiguration to: MyAbstractConfiguration
+```
+
+### Collect pragmas in the image
+
+The second way to collect the pragmas to handle them is to use the PragmaCollector. This class can be configured with a query and will return all the pragmas matching the query.
+
+To configure the PragmaCollector we need to send implement a filter block. If we want to collect the primitive pragmas:
+
+```Smalltalk
+PragmaCollector filter: [:prg | prg keyword = 'primitive:' ]
+```
+
+Once configured we can execute the collection of the pragmas by sending the method #reset to the pragma collector:
+
+```Smalltalk
+(PragmaCollector filter: [:prg | prg keyword = 'primitive:' ]) reset
+```
+
+Then you can iterate on the result using #do:, #collect:, #reject:, etc or using them directly through #collected. 
+
+In our example we might want to access them this way:
+
+```Smalltalk
+(PragmaCollector filter: [:prg | prg keyword = 'applicationParameterOptional:' ]) reset collected
+```
+
+Or we can select only non optional parameters this way:
+
+```Smalltalk
+(PragmaCollector filter: [:prg | prg keyword = 'applicationParameterOptional:' and: [ (prg argumentAt: 1) not ] ]) reset collected
+```
+
+## Act on collected pragmas
+
+Once collected, we need to act on pragmas.
```

Comment:
What do you mean by act?

Reply:
Use them to implement our feature. I'll clarify it in a following commit.